rcmd/helpers.go

```diff
@@ -5,32 +5,43 @@ import (
 	"path/filepath"
 	"runtime"
 	"strings"
+
+	log "github.com/sirupsen/logrus"
 )
 
-func binaryName(pkg, version string) string {
-	switch runtime.GOOS {
+func binaryNameOs(os, pkg, version, platform string) string {
+	switch os {
 	case "darwin":
 		return fmt.Sprintf("%s_%s.tgz", pkg, version)
 	case "linux":
-		return fmt.Sprintf("%s_%s_R_x86_64-pc-linux-gnu.tar.gz", pkg, version)
+		return fmt.Sprintf("%s_%s_R_%s.tar.gz", pkg, version, platform)
 	case "windows":
 		return fmt.Sprintf("%s_%s.zip", pkg, version)
 	default:
-		fmt.Println("platform not supported for binary detection")
-		return ""
+		log.Fatal("platform not supported for binary detection")
+		return ("")
```

Comment:
no need for a return here as same with fatal 

have a empty return after the switch for the compiler happiness